apollo-router-core/src/naive_introspection.rs

```diff
@@ -24,22 +25,24 @@ static KNOWN_INTROSPECTION_QUERIES: Lazy<Vec<String>> = Lazy::new(|| {
 });
 
 /// A cache containing our well known introspection queries.
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub struct NaiveIntrospection {
-    cache: HashMap<String, Response>,
+    cache: RwLock<HashMap<String, Response>>,
 }
 
 impl NaiveIntrospection {
     #[cfg(test)]
     pub fn from_cache(cache: HashMap<String, Response>) -> Self {
-        Self { cache }
+        Self {
+            cache: RwLock::new(cache),
+        }
     }
 
     /// Create a `NaiveIntrospection` from a `Schema`.
     ///
     /// This function will populate a cache in a blocking manner.
     /// This is why `NaiveIntrospection` instanciation happens in a spawn_blocking task on the state_machine side.
-    pub fn from_schema(schema: &Schema) -> Self {
+    pub async fn from_schema(schema: &Schema) -> Self {
```

Comment:
You should remove lines 43 and 44 now this is `async` and you have removed the `spawn_blocking`. (or update it to something more accurate)